dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/models/DLCAnnouncementDAO.scala

```diff
@@ -0,0 +1,118 @@
+package org.bitcoins.dlc.wallet.models
+
+import org.bitcoins.crypto._
+import org.bitcoins.db._
+import org.bitcoins.dlc.wallet.DLCAppConfig
+import slick.lifted._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+case class DLCAnnouncementDAO()(implicit
+    val ec: ExecutionContext,
+    override val appConfig: DLCAppConfig)
+    extends CRUD[DLCAnnouncementDb, (Sha256Digest, Long)]
+    with SlickUtil[DLCAnnouncementDb, (Sha256Digest, Long)] {
+  private val mappers = new org.bitcoins.db.DbCommonsColumnMappers(profile)
+  import mappers._
+  import profile.api._
+
+  override val table: TableQuery[DLCAnnouncementTable] =
+    TableQuery[DLCAnnouncementTable]
+
+  private lazy val announcementDataTable: slick.lifted.TableQuery[
+    OracleAnnouncementDataDAO#OracleAnnouncementsTable] = {
+    OracleAnnouncementDataDAO().table
+  }
+
+  private lazy val dlcTable: slick.lifted.TableQuery[DLCDAO#DLCTable] = {
+    DLCDAO().table
+  }
+
+  override def createAll(
+      ts: Vector[DLCAnnouncementDb]): Future[Vector[DLCAnnouncementDb]] =
+    createAllNoAutoInc(ts, safeDatabase)
+
+  override protected def findByPrimaryKeys(
+      ids: Vector[(Sha256Digest, Long)]): profile.api.Query[
+    DLCAnnouncementTable,
+    DLCAnnouncementDb,
+    Seq] = {
+
+    // is there a better way to do this?
+    val starting = table.filterNot(_.dlcId === Sha256Digest.empty)
```

Comment:
This is weird, I agree. I couldn't put just `table` for the init in the `foldLeft` so I instead just did a filter that shouldn't filter anything.

Reply:
I think it can just be implemented like this ? 

```scala
    val dlcIds = ids.map(_._1)
    val announcementId = ids.map(_._2)
    table.filter(t =>
      t.dlcId.inSet(dlcIds) && t.announcementId.inSet(announcementId))
```